### ISSUE TITLE
Behaviour of "runAfter" for a Task that runs after a Task with Condition

### ISSUE THREAD

**Issue by afrittoli** [tektoncd/pipeline#1264]
# Expected Behavior
I would like to have a way in Tekton to easily express the following dependency between Tasks in a pipeline: The 2nd task should run only after the 1st one was executed successfully or if the 1st one was skipped because of a condition.

# Actual Behavior
If the 1st task is skipped because the condition is not met, the 2nd task is not executed at all.

# Steps to Reproduce the Problem

1. Create a pipeline like similar to this:
```
apiVersion: tekton.dev/v1alpha1
kind: Pipeline
metadata:
  name: two-tasks-one-condition
spec:
  tasks:
  - name: conditional-task
    taskRef:
      name: task1
    conditions:
      - conditionRef: always-fail
  - name: run-if-1-or-1-skipped
    taskRef:
      name: task2
    runAfter:
      - conditional-task
```
2. Run the pipeline through a PipelineRun
3. task2 is not executed

# Additional Info
In the example pipeline I used `runAfter` but I believe the behaviour would be the same if the dependency was introduced using `from` on a resource.

**Comment by dibyom** [comment]
This is a feature request that looks like a duplicate of #1023 

I'm going to close this out though I like the description here so I copied the `Steps to Reproduce the Problem` section over to #1023 :angel:
